assembly/src/parsers/io_ops.rs

```diff
@@ -26,7 +28,39 @@ pub fn parse_push(op: &Token) -> Result<Node, ParsingError> {
     validate_operation!(op, "push", 1..MAX_PUSH_INPUTS);
 
     let constants = parse_constants(op)?;
-    Ok(Instruction(PushConstants(constants)))
+
+    if constants.len() == 1 {
+        let value = constants[0].as_int();
+        if value <= u8::MAX as u64 {
+            Ok(Instruction(PushU8(value as u8)))
+        } else if value <= u16::MAX as u64 {
+            Ok(Instruction(PushU16(value as u16)))
+        } else if value <= u32::MAX as u64 {
+            Ok(Instruction(PushU32(value as u32)))
+        } else {
+            Ok(Instruction(PushFelt(constants[0])))
+        }
+    } else {
+        let constants_u64: Vec<u64> = constants.iter().map(|v| v.as_int()).collect();
+        match constants_u64.iter().max() {
```

Comment:
Why do we have to collect to then fetch the max value? I think we can just call `max` directly there

Reply:
Indeed, it is not necessary. My bad